### PR TITLE
[Bug] Fix run away ability

### PR DESCRIPTION
**Pull request by SeafoamQueen**
## What are the changes?
Pokemon with the ability Run Away are not able to flee from battles where an opposing pokemon has a trapping ability.

## Why am I doing these changes?
A Pokémon with the Run Away ability can always flee and use Teleport successfully, regardless of trapping moves and abilities.
- https://bulbapedia.bulbagarden.net/wiki/Run_Away_(Ability)#Effect

## What did change?
`ArenaTrapAbAttr` in `ability.ts` was modified to add a check for the ability `Run_Away`.

### Screenshots/Videos
The attached video shows a pokemon with the ability Run Away successfully running from pokemon with the ability Arena Trap and the passive Shadow Tag.

https://github.com/pagefaultgames/pokerogue/assets/167576411/c8c3981d-3c33-4cf1-a696-f2e66d8d9af9

## How to test the changes?
Start a new game with a pokemon with the ability Run Away.
Set the following overrides in `overrides.ts` :
- `OPP_ABILITY_OVERRIDE` to `Abilities.ARENA_TRAP` or `Abilities.SHADOW_TAG`

Using each pokemon attempt to run from battles.
You should successfully escape everytime, unless you are in a trainer battle in Classic or an x50 battle in Endless.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by DecaP0d** [comment]
As I've said before, I'm heavily against the implementation of Ghost types being given built-in Run Away. This might be something that works for the main series games, but I feel is harmful for Pokérogue. The choice to use a Run Away pokémon should come at a cost due to its potential to bypass fights that could end your run, and most pokémon that have the ability are either weak or have to give up an ability slot for it. For future modes that pose any challenge, this change will stop any Run Away passives from being added and will create a bad dynamic where players will want to always have a ghost type on their team. Please reconsider this! The change to how Run Away interacts with Shadow Tag and Arena Trap is great though.

File: src/phases.ts [hunk 1]
```diff
@@ -2199,7 +2199,16 @@ export class TurnStartPhase extends FieldPhase {
             const fasterPokemon = playerActivePokemon[0].getStat(Stat.SPD) > playerActivePokemon[1].getStat(Stat.SPD) ? playerActivePokemon[0] : playerActivePokemon[1];
             // check if either active pokemon has the ability "Run Away"
             const hasRunAway = playerActivePokemon.find(p => p.hasAbility(Abilities.RUN_AWAY));
-            runningPokemon = hasRunAway !== undefined ? hasRunAway : fasterPokemon;
+            // check if either active pokemon has the Ghost type
+            const hasGhostType = playerActivePokemon.find(p => p.getTypes(true).includes(Type.GHOST) || (p.getTypes(true).includes(Type.STELLAR) && p.getTypes().includes(Type.GHOST)));
```
Comment (brain-frog) [inline src/phases.ts:2203]: is the stellar and ghost case necessary? won't getting ghost on its own cover that?

Comment (SeafoamQueen) [reply src/phases.ts:2203]: I thought the same thing, but did it this way based on the way this PR to finish Magnet Pull and Arena Trap handled ghost types.
https://github.com/pagefaultgames/pokerogue/pull/1193

**Comment by SeafoamQueen** [comment]
@brain-frog The ghost type mechanics have been removed from the PR. It should be fine to merge now.

**Comment by SeafoamQueen** [comment]
> What about `Magnet Pull` ?

Magnet Pull shouldn't trap pokemon with the ability Run Away either, and it uses the same `ArenaTrapAbAttr` property that the other trapping abilities use to determine if a pokemon is trapped by the ability.
I've modified that property to check for the ability Run Away, and not trap the pokemon if it has it.

**Review by brain-frog — APPROVED** [review]

**Pull request merged**